sdk/dotnet/Pulumi/Resources/ComponentResourceOptions.cs

```diff
@@ -55,13 +56,10 @@ public static ComponentResourceOptions Merge(ComponentResourceOptions? options1,
             options1 = options1 != null ? CreateComponentResourceOptionsCopy(options1) : new ComponentResourceOptions();
             options2 = options2 != null ? CreateComponentResourceOptionsCopy(options2) : new ComponentResourceOptions();
 
-            ExpandProviders(options1);
-            ExpandProviders(options2);
-
             // first, merge all the normal option values over
             MergeNormalOptions(options1, options2);
 
-            options1.Providers.AddRange(options2.Providers);
+            options1.Providers = MergeProviders(options1.Providers, options2.Providers);
 
             if (options1.Providers.Count == 1)
```

Comment:
Shouldn't this go as well? I thought we weren't promoting up anymore.

Reply:
My bad requesting a review. I'm not done with this PR yet.

Reply:
I mean lines 64-68 in the if statement. It's setting `Provider` if `Providers.Count == 1`

Reply:
Also no worries, quick early reviews don't hurt :)